### PR TITLE
feat(charter): introduce emergency and async votes

### PR DESCRIPTION
**Pull request by mmarchini**
Asynchronous voting is appropriate for issues that should be votes by
as many TSC members as possible.

Emergency votes are intended as a tool to resolve issues that are
time-sensitive and TSC can't reach consensus on it in a timely manner
asynchronously.

### CODE REVIEW

**Comment by mcollina** [comment]
I would prefer to call them "immediate votes".

**Comment by mmarchini** [comment]
I went with "emergency" because I thought it was a more familiar terminology, but I'm happy with "immediate" as well

File: TSC-Charter.md
Comment (targos) [inline TSC-Charter.md:158]:
```suggestion
issue will be discussed in the next meeting (where a regular vote can be
```

**Comment by MylesBorins** [comment]
Quick note that charter changes will need to be approved by the CPC. One thought. Could we maybe make the language in the charter around votes more ambiguous and move the specifics of implementation to the governance? That way if we want to iterate on this new process we can do so without having to go back-and-forth with the CPC

File: TSC-Charter.md
Comment (mhdawson) [inline TSC-Charter.md:152]: My personal feeling is it might be better to just allow votes to be called through the mailing list.  I'm not sure a meeting is needed, and the mailing is likely to reach more TSC members.

Comment (MylesBorins) [reply TSC-Charter.md:152]: +1 to making a clear process for calling a vote without requiring a meeting rather than making a specific carve out. Honestly if one TSC member calls the vote and there are no objections within a 24 hour period that could be reasonable enough to kick one off. Waiting for a majority of the TSC to +1 could end up causing a delay long enough that we have the next meetings ready to go already 😂

Comment (mmarchini) [reply TSC-Charter.md:152]: So instead of having two ways of calling for vote and executing a vote, we only have one via mailing list (potentially via issues for votes that can be public)? I think that's a good idea, it allows for input from any TSC members (not only those who live in a timezone favourable for a particular meeting) and allow us to vote faster.

Comment (mmarchini) [reply TSC-Charter.md:152]: @MylesBorins +100000, there's no reason to wait for 10 +1s via email to start the vote process, 24h with no one objecting to having a vote seems reasonable.

Comment (MylesBorins) [reply TSC-Charter.md:152]: If someone doesn't like the vote they can vote no 😂

Comment (mcollina) [reply TSC-Charter.md:152]: I agree! I would like to see the need for calling a vote in a meeting or with a majority removed.

**Comment by Trott** [comment]
Just to make sure everyone is aware: Changes to the charter need to be approved by the foundation board. So let's figure out what we want soon because there's a second set of people that need to concur!

**Comment by Trott** [comment]
> Quick note that charter changes will need to be approved by the CPC.

Oh, I thought it was the board. CPC makes more sense, I think.

**Comment by MylesBorins** [comment]
@Trott it use to be the board until the OpenJS foundation was formed. After OpenJS started the CPC was chartered with the responsibility of overseeing project charters and the Node.js TSC + CommComm were rechartered under the CPC.

IMHO this is a massive benefit to the new structure as we have far more representation on the CPC and they meet far more regularly.

**Comment by mhdawson** [comment]
@mmarchini do you want to update this PR to reflect what seems to be the consensus?

**Comment by mmarchini** [comment]
yes, i just haven't had time to do it yet. If someone else wants to do it though, feel free to do so

**Comment by mmarchini** [comment]
Ok, I made the following changes:

- Call for votes follows Lazy Consensus now
- Renamed "Emergency Votes" to "Immediate Votes"
- Calling an immediate vote must be done in the issue tracker, and not via mailing list (so the call is public)

While I like the idea of merging both votes and moving more vote details out of the charter to our governance, I think it can be done in another PR. This PR should considerably improve the situation already.

**Comment by mmarchini** [comment]
cc @nodejs/tsc as we need majority approval for charter changes. Once we approved it at TSC level I'll raise it to CPC.

File: TSC-Charter.md
Comment (mcollina) [inline TSC-Charter.md:155]: There have been a few occasions where the issue to vote on was private in nature. In those cases, we should use emails.

File: TSC-Charter.md
Comment (mhdawson) [inline TSC-Charter.md:144]:
```suggestion
a closing vote (assumes enough TSC members in the meeting), a vote to table the
issue to the next meeting, or a vote to be conducted asynchronously through a
system to be chosen by the TSC. The call for a vote is approved if none of the
TSC members in the meeting object to it.
```

**Review by mhdawson — APPROVED** [review]
LGTM

File: TSC-Charter.md
Comment (apapirovski) [inline TSC-Charter.md:152]: Wouldn't time-sensitive be more apt (already used in describing the process below)? The phrase `immediate` seems pretty confusing to me, especially since the process is still not truly immediate.

Comment (mcollina) [reply TSC-Charter.md:152]: Yes

**Review by joesepi — APPROVED** [review]

**Comment by bnb** [comment]
+1. Don't want to approve directly since it'll green check and this isn't a space where I should be green checking ❤️

**Review by WaleedAshraf — APPROVED** [review]

**Comment by bnb** [comment]
Any progress on this?

**Comment by bnb** [comment]
removed the cc-agenda label for now. Feel free to add it back for us when this is active/ready to merge again <3

**Comment by Trott** [comment]
Given that this has been open for more than a year and has seen no activity for more than 7 months, I'm going to close this. But if you feel that's the wrong thing to do and there's still a need for pursuing this change, please re-open. Thanks!

**Pull request closed without merging**
